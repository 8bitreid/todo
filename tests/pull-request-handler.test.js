const { gimmeRobot, loadConfig, loadDiff } = require('./helpers')
const pullRequestOpened = require('./fixtures/payloads/pull_request.opened.json')

describe('pull-request-handler', () => {
  let robot, github
  const event = { event: 'pull_request', payload: pullRequestOpened }

  beforeEach(() => {
    const gimme = gimmeRobot()
    robot = gimme.robot
    github = gimme.github
  })

  it('comments on a pull request', async () => {
    await robot.receive(event)
    expect(github.issues.createComment).toHaveBeenCalledTimes(1)
    expect(github.issues.createComment.mock.calls[0]).toMatchSnapshot()
  })

  it('comments on a pull request and mentions the assigned user', async () => {
    github.repos.getContent.mockReturnValueOnce(loadConfig('autoAssignString'))
    await robot.receive(event)
    expect(github.issues.createComment.mock.calls[0]).toMatchSnapshot()
  })

  it('comments on a pull request and mentions the assigned users', async () => {
    github.repos.getContent.mockReturnValueOnce(loadConfig('autoAssignArr'))
    await robot.receive(event)
    expect(github.issues.createComment.mock.calls[0]).toMatchSnapshot()
  })

  it('does not create duplicate comments', async () => {
    github.issues.getComments.mockReturnValueOnce(Promise.resolve({ data: [{
      body: '## I am an example title'
    }] }))

    await robot.receive(event)
    expect(github.issues.createComment).toHaveBeenCalledTimes(0)
  })

  it('creates many (5) comments', async () => {
    github.pullRequests.get.mockReturnValueOnce(loadDiff('many'))
    await robot.receive(event)
    expect(github.issues.createComment).toHaveBeenCalledTimes(5)
  })
})
