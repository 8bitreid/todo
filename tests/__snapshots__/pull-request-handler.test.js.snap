// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pr-comment-handler comments on a pull request 1`] = `
Array [
  Object {
    "body": "## I am an example title

https://github.com/JasonEtco/tests/blob/1409a3f61a27fe80a7170026fd73eaa2108a0108/.travis.yml#L11

---

###### This comment was generated by [todo](https://todo.jasonet.co) based on a \`TODO\` comment in 1409a3f61a27fe80a7170026fd73eaa2108a0108 in #32. cc @JasonEtco.",
    "number": 32,
    "owner": "JasonEtco",
    "repo": "tests",
  },
]
`;

exports[`pr-comment-handler comments on a pull request and mentions the assigned user 1`] = `
Array [
  Object {
    "body": "## I am an example title

https://github.com/JasonEtco/tests/blob/1409a3f61a27fe80a7170026fd73eaa2108a0108/.travis.yml#L11

---

###### This comment was generated by [todo](https://todo.jasonet.co) based on a \`TODO\` comment in 1409a3f61a27fe80a7170026fd73eaa2108a0108 in #32. cc @matchai",
    "number": 32,
    "owner": "JasonEtco",
    "repo": "tests",
  },
]
`;

exports[`pr-comment-handler comments on a pull request and mentions the assigned users 1`] = `
Array [
  Object {
    "body": "## I am an example title

https://github.com/JasonEtco/tests/blob/1409a3f61a27fe80a7170026fd73eaa2108a0108/.travis.yml#L11

---

###### This comment was generated by [todo](https://todo.jasonet.co) based on a \`TODO\` comment in 1409a3f61a27fe80a7170026fd73eaa2108a0108 in #32. cc @JasonEtco, @matchai and @defunkt",
    "number": 32,
    "owner": "JasonEtco",
    "repo": "tests",
  },
]
`;
